Upstream-Status: Submitted
Signed-off-by: Ross Burton <ross.burton@intel.com>

commit 1affe20d5c82befc3b1626e557409dab5343c47b
Author: Ross Burton <ross.burton@intel.com>
Date:   Mon Jul 7 12:53:35 2014 +0100

    configure.ac: add option for xshmfence
    
    xshmfence is usable outside of DRI3, and is currently autodetected which isn't
    good for distributions where deterministic builds are aspired to.
    
    Signed-off-by: Ross Burton <ross.burton@intel.com>

diff --git a/configure.ac b/configure.ac
index 2daa6be..fd1cf3f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -648,6 +648,7 @@ AC_ARG_ENABLE(kdrive-kbd,     AS_HELP_STRING([--enable-kdrive-kbd], [Build kbd d
 AC_ARG_ENABLE(kdrive-mouse,   AS_HELP_STRING([--enable-kdrive-mouse], [Build mouse driver for kdrive (default: auto)]), [KDRIVE_MOUSE=$enableval], [KDRIVE_MOUSE=auto])
 AC_ARG_ENABLE(kdrive-evdev,   AS_HELP_STRING([--enable-kdrive-evdev], [Build evdev driver for kdrive (default: auto)]), [KDRIVE_EVDEV=$enableval], [KDRIVE_EVDEV=auto])
 AC_ARG_ENABLE(libunwind,      AS_HELP_STRING([--enable-libunwind], [Use libunwind for backtracing (default: auto)]), [LIBUNWIND="$enableval"], [LIBUNWIND="auto"])
+AC_ARG_ENABLE(xshmfence,      AS_HELP_STRING([--disable-xshmfence], [Disable xshmfence (default: auto)]), [WANT_XSHMFENCE="$enableval"], [WANT_XSHMFENCE="auto"])
 
 
 dnl chown/chmod to be setuid root as part of build
@@ -1235,18 +1236,19 @@ esac
 
 AM_CONDITIONAL(BUSFAULT, test x"$BUSFAULT" = xyes)
 
-PKG_CHECK_MODULES([XSHMFENCE], $XSHMFENCE,
-		  [HAVE_XSHMFENCE=yes], [HAVE_XSHMFENCE=no])
+HAVE_XSHMFENCE=no
+if test "x$WANT_XSHMFENCE" != "xno"; then
+	PKG_CHECK_MODULES([XSHMFENCE], $XSHMFENCE,
+			  [HAVE_XSHMFENCE=yes], [HAVE_XSHMFENCE=no])
 
+	if test "$WANT_XSHMFENCE,$HAVE_XSHMFENCE" = "yes,no"; then
+		AC_MSG_ERROR([libxshmfence requested but not found.])
+	fi
+	AC_DEFINE(HAVE_XSHMFENCE, 1, [Have X Shared Memory Fence library])
+	REQUIRED_LIBS="$REQUIRED_LIBS xshmfence"
+fi
 AM_CONDITIONAL(XSHMFENCE, test "x$HAVE_XSHMFENCE" = xyes)
 
-case x"$HAVE_XSHMFENCE" in
-	xyes)
-		AC_DEFINE(HAVE_XSHMFENCE, 1, [Have X Shared Memory Fence library])
-		REQUIRED_LIBS="$REQUIRED_LIBS xshmfence"
-		;;
-esac
-
 
 case "$DRI3,$HAVE_XSHMFENCE" in
 	yes,yes | auto,yes)
